src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -568,7 +572,12 @@ int ReadBuffered()
 
             protected override void Dispose(bool disposing)
             {
-                _reader?.Dispose();
+                if (_reader != null && !_reader.IsDisposed)
+                {
+                    _reader.Invoke("cancel");
+                    _reader.Dispose();
+                    _reader = null;
+                }
                 _status?.Dispose();
```

Comment:
This now calls into `AbortController.abort()`, right? Is it necessary to call `ReadableStream.cancel()` in addition to this?

Reply:
Good question. At the point when we are already reading from the stream, we are no longer awaiting the `fetch`. We already received the HTTP headers and now, we are reading the stream. I could not find any hint of how `AbortController` influences `ReadableStream` in the documentation.

Is there downside to doing both ?

Reply:
I tested that both FF and Chrome will stop streaming with both `AbortController.abort()` and `ReadableStream.cancel()`

Reply:
Since we are disposing the `WasmFetchResponse` during `WasmHttpReadStream.Dispose` the `abort()` would be called there. 
But the new tests uncovered scenarios, when `ReadAsync` didn't handle `cancellationToken` well. In such cases I use `cancel()` now.